tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs

```diff
@@ -391,32 +380,34 @@ private async Task AddAllComputeSystemsFromAProvider(ComputeSystemsLoadedData da
     [RelayCommand]
     public void SearchHandler(string query)
     {
-        var currentProviders = PerProviderViewModels.ToList();
-        PerProviderViewModels.Clear();
-
-        var dontShowIndices = new List<int>();
-        for (var i = 0; i < currentProviders.Count; i++)
+        ComputeSystemCardsView.Filter = system =>
         {
-            var providerViewModel = currentProviders[i];
-            providerViewModel.SearchHandler(query);
-
-            if (!providerViewModel.IsVisible)
+            if (system is CreateComputeSystemOperationViewModel createComputeSystemOperationViewModel)

```

Comment:
<Not for this PR as it's a revert of a previous change>
I think we abuse easiness that C# gives us to determine variable type at runtime (reflection) and use in places where we could do a better design instead. In this case we could have an interface that "system" would be required to implement in order to be used for search queries. It's "system's" job to know how to match that query. Then the code here would just be "return system.Match(_selectedProvider, query);"